app/controllers/helpers/DeclarationHolderHelper.scala

```diff
@@ -16,13 +16,43 @@
 
 package controllers.helpers
 
+import forms.declaration.additionaldeclarationtype.AdditionalDeclarationType._
 import forms.declaration.declarationHolder.DeclarationHolder
 import models.requests.JourneyRequest
+import play.api.data.FormError
 
 object DeclarationHolderHelper {
 
   val DeclarationHolderFormGroupId: String = "declarationHolder"
 
   def cachedHolders(implicit request: JourneyRequest[_]): Seq[DeclarationHolder] =
     request.cacheModel.parties.declarationHoldersData.map(_.holders).getOrElse(Seq.empty)
+
+  def validateAuthCode(maybeDeclarationHolder: Option[DeclarationHolder])(implicit r: JourneyRequest[_]): Option[FormError] =
+    maybeDeclarationHolder match {
+      case Some(DeclarationHolder(Some(authorisationCode), _, _)) =>
+        if (isExrrSelectedForPrelodgedDecl(authorisationCode, r.cacheModel.additionalDeclarationType))
+          Some(FormError(DeclarationHolderFormGroupId, "declaration.declarationHolder.EXRR.error.prelodged"))
+        else
+          validateMutuallyExclusiveAuthCodes(authorisationCode, cachedHolders)
+
+      case _ => None
+    }
+
+  private val nonExrrAdditionalDeclarationTypes =
+    List(STANDARD_PRE_LODGED, SIMPLIFIED_PRE_LODGED, OCCASIONAL_PRE_LODGED, CLEARANCE_PRE_LODGED)
+
+  private def isExrrSelectedForPrelodgedDecl(authorisationCode: String, declarationType: Option[AdditionalDeclarationType]): Boolean =
+    declarationType.fold(false)(authorisationCode == "EXRR" && nonExrrAdditionalDeclarationTypes.contains(_))
+
+  private val mutuallyExclusiveAuthorisationCodes = List("CSE", "EXRR")
+
+  private def validateMutuallyExclusiveAuthCodes(authorisationCode: String, holders: Seq[DeclarationHolder]): Option[FormError] =
+    if (!mutuallyExclusiveAuthorisationCodes.contains(authorisationCode)) None
+    else {
+      val mustNotAlreadyContainCodes = mutuallyExclusiveAuthorisationCodes.filter(_ != authorisationCode)
+
+      if (!holders.map(_.authorisationTypeCode.getOrElse("")).containsSlice(mustNotAlreadyContainCodes)) None
+      else Some(FormError(DeclarationHolderFormGroupId, s"declaration.declarationHolder.${authorisationCode}.error.exclusive"))
+    }
```

Comment:
This is a kind of validation, so I'd place it in the `Form` class and this is what we used to do in past.
This way there is no need to call it explicitly in addition to regular form binding.